### PR TITLE
ci: added commit lint, build check, test and release

### PR DESCRIPTION
**Pull request by vojtechsimetka**
Adds CI running commit lint, building the project, running test and action creating realease PR with autogenerated changelogs. To get the tests running I had to update several of the node modules and fix compilation errors.

NOTE: code linting is not part of this PR. That would require much larger code change which I want to keep separate.

### CODE REVIEW

File: src/components/CodeBlock.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import SyntaxHighlighter from 'react-syntax-highlighter';
-import { hybrid } from 'react-syntax-highlighter/dist/esm/styles/hljs';
+// import { hybrid } from 'react-syntax-highlighter/dist/esm/styles/hljs';
```
Comment (agazso) [inline src/components/CodeBlock.tsx:2]: Please do not commit commented code

File: src/pages/status/NodeSetupWorkflow.tsx [hunk 2]
```diff
@@ -55,45 +55,47 @@ function getStepContent(step: number, props: any) {
...
   useEffect(() => {
+    const handleComplete = (index: number) => {
+      const newCompleted = completed;
+      newCompleted[index] = true;
+      setCompleted(newCompleted);
+    };
+
+    const evaluateNodeStatus = () => {
+      if (nodeHealth?.status === 'ok' && nodeApiHealth) {
+        handleComplete(0)
```
[the remark this answers is not in the document]
Comment (vojtechsimetka) [reply src/pages/status/NodeSetupWorkflow.tsx:73]: Yeah but I don't want to do it in this PR. Thats not the scope. There is quite a lot of code like that.

File: src/pages/status/SetupSteps/ChequebookDeployFund.tsx [hunk 3]
```diff
@@ -26,7 +25,7 @@ export default function ChequebookDeployFund(props: any) {
                 :
                 <div>
                     <Warning style={{color:'#ff9800', marginRight: '7px', height: '18px'}} />
-                    <span>Your chequebook is either not deployed or funded. Run the below commands to get your address and deposit ETH. Then visit the BZZaar here <a href='#'>https://bzz.ethswarm.org/?transaction=buy&amount=10&slippage=30&receiver=[ENTER_ADDRESS_HERE]</a> to get BZZ</span>
+                    <span>Your chequebook is either not deployed or funded. Run the below commands to get your address and deposit ETH. Then visit the BZZaar here <Typography variant='button'>https://bzz.ethswarm.org/?transaction=buy&amount=10&slippage=30&receiver=[ENTER_ADDRESS_HERE]</Typography> to get BZZ</span>
```
Comment (agazso) [inline src/pages/status/SetupSteps/ChequebookDeployFund.tsx:28]: I wonder if the address could not be filled automatically.

Comment (vojtechsimetka) [reply src/pages/status/SetupSteps/ChequebookDeployFund.tsx:28]: Yeah I wondered too, I'll create an issue about it and address it separately.

**Pull request merged**